### PR TITLE
front: fix various problems related to long rolling stock names

### LINKED ISSUE #7278: non-responsive display for a rolling stock with a long name
### What happened?

When creating a rolling stock with a long name , subseries and details ... the created field is not correctly viewed 
image

### What did you expect to happen?

Add ellipsis for fields with long value 

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to rolling stock editor
2. Create a new rolling stock with long value in the different fields
3. Verify the display of the fields 

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: W11
- **Env**: rec

### OSRD version (top right corner `Account` button > `Informations`)

1482e8c

### LINKED ISSUE #8439: Long rolling stock name is tightening other tabs during train creation
### What happened?

image

### What did you expect to happen?

An ellipsis should be fine

### How can we reproduce it (as minimally and precisely as possible)?

1. Try a very long rolling stock name in the train creation window

### On which environments the bug occurs?

Dev (SNCF)

### On which browser the bug occurs?

Google Chrome

### OSRD version (top right corner `Account` button > `Informations`)

0f97f32f7509cc9722a1a33bec7462ee22009e82

### PR DESCRIPTION
**Pull request by clarani**
closes #8439 
closes #7278

image

### CODE REVIEW

File: front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss [hunk 1]
```diff
@@ -3,9 +3,11 @@
     display: flex;
     align-items: center;
     width: 100%;
+    overflow: hidden;
+    text-overflow: ellipsis;
```
Comment (SharglutDev) [inline front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss]: Do we need these with the one in `.rolling-stock-name` class ?

File: front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss [hunk 2]
```diff
@@ -16,6 +18,14 @@
         -webkit-mask-image: linear-gradient(90deg, black 80%, rgba(0, 0, 0, 0));
       }
     }
+
+    .rolling-stock-name {
+      margin-left: 8px;
+      width: calc(100% - 48px - 8px); // 48px for the rs image width, 8px for the left margin
```
Comment (SharglutDev) [inline front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss:23]: Do we keep magic numbers (perfectly fine with your comment) or do we create css variables in the file to avoid the comment ?

**Review by SharglutDev — COMMENTED** [review]
Great fix thanks ! Left just some nit comments.

Also, we open the pandora box with this issue :upside_down_face: 

Can't see the whole rs name
image

Weird style in the rs card
Capture d’écran du 2025-02-19 17-55-07

Horizontal scroll if the rs name is too long (hiding the action buttons)
Capture d’écran du 2025-02-19 17-53-44

File: front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss (hunk 2 above)
Comment (clarani) [reply front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss:23]: I think it's ok to keep the magic numbers here since they are explained in the comment :)

Comment (SharglutDev) [reply front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss:23]: Fine by me :)

**Comment by clarani** [comment]
I fixed some of the issues:
- the trains are flipped in the timetable cards
  image

- the names are not wrapped anymore
  image

- ellipsis were added in the rolling stock editor as well
  image

I won't go further since these components will soon be removed, if it's fine for you :+1:

File: front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss (hunk 1 above)
Comment (clarani) [reply front/src/styles/scss/applications/operationalStudies/_managetrainschedule.scss]: Fixed :)

**Comment by theocrsb** [comment]
I have the impression that there is still some strange behavior when the train name is really long : 
image

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, thank you for these painfull fixes :)
(There are still some weird edge cases but it's unlikely that they happen in a real life use of the app)

**Comment by clarani** [comment]
@theocrsb can you test again ? I can't reproduce anymore

**Review by theocrsb — APPROVED** [review]
Sounds good to me. Paul is right about the other use cases.
